WorkspaceServer/Kernel/EmitProcessors.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Scripting;
+
+namespace WorkspaceServer.Kernel
+{
+    public class EmitProcessors : ICodeSubmissionProcessor
+    {
+        private readonly Func<ScriptState> _getScriptState;
+
+        public EmitProcessors(Func<ScriptState> getScriptState)
+        {
+            _getScriptState = getScriptState;
+            Command = new Command("emit");
+            var outputOption = new Option("--output")
+            {
+                Argument = new Argument<DirectoryInfo>()
+            };
+            
+            Command.AddOption(outputOption);
+            Command.Handler = CommandHandler.Create<EmitProcessorsOptions>((options) =>
+            {
+                if (!options.Output.Exists)
+                {
+                    options.Output.Create();
+                }
+                
+                var state = getScriptState();
+
+                var codeFile = new FileInfo(Path.Combine(options.Output.FullName, "code.cs"));
```

Comment:
>Path.Combine(options.Output.FullName, "code.cs") [](start = 44, length = 48)

This should be made to work against IDirectoryAccessor from the outset.

Reply:
indeed